run.py

```diff
@@ -248,23 +257,139 @@ def run(args):
+
+        if os.environ.get("TOOL") is not None:
+            ci_message = json.loads(os.environ["CI_MESSAGE"])
+            compose_id = ci_message["compose_id"]
+            compose_url = ci_message["compose_url"] + "/"
+            product_name = ci_message.get("product_name", None)
+            product_version = ci_message.get("product_version", None)
+            log.info("COMPOSE_URL = %s ", compose_url)
+
+            if os.environ["TOOL"] == "pungi":
+                # is a rhel compose
+                log.info("trigger on CI RHEL Compose")
+            elif os.environ["TOOL"] == "rhcephcompose":
+                # is a ubuntu compose
+                log.info("trigger on CI Ubuntu Compose")
+                ubuntu_repo = compose_url
+                log.info("using ubuntu repo" + ubuntu_repo)
+            elif os.environ["TOOL"] == "bucko":
+                # is a docker compose
+                log.info("Trigger on CI Docker Compose")
+                docker_registry, docker_tag = ci_message["repository"].split(
+                    "/rh-osbs/rhceph:"
+                )
+                docker_image = "rh-osbs/rhceph"
+                log.info(
+                    f"\nUsing docker registry from ci message: {docker_registry} \n"
+                    f"Docker image: {docker_image}\nDocker tag:{docker_tag}"
+                )
+                log.warning("Using Docker insecure registry setting")
+                docker_insecure_registry = True
+
+            if product_name == "ceph":
+                # is a rhceph compose
+                base_url = compose_url
+                log.info("using base url" + base_url)
+
+        if not os.environ.get("TOOL") and not ignore_latest_nightly_container:
+            try:
+                latest_container = get_latest_container(rhbuild)
+            except ValueError:
+                print(
+                    "ERROR:No latest nightly container UMB msg at "
+                    "/ceph/cephci-jenkins/latest-rhceph-container-info/ "
+                    "specify using the cli args or use --ignore-latest-container"
+                )
+                sys.exit(1)
+            docker_registry = (
+                latest_container.get("docker_registry")
+                if not docker_registry
+                else docker_registry
+            )
+            docker_image = (
+                latest_container.get("docker_image")
+                if not docker_image
+                else docker_image
+            )
+            docker_tag = (
+                latest_container.get("docker_tag") if not docker_tag else docker_tag
+            )
+            log.info(
+                f"Using latest nightly docker image - {docker_registry}/{docker_image}:{docker_tag}"
+            )
+            docker_insecure_registry = True
+            log.warning("Using Docker insecure registry setting")
+    else:
+        platform = args.get("--platform", None)
```

Comment:
Can we default to `rhel-8`?

Reply:
I thought not to fall back on default OS version which leads handling unnecessary dependents.
- RHCS compose can be picked for wrong OS platform if we did not handle inventory, Which actually fails at Test case rather than early.

However, We can go through "rhel-8" unless we provided wrong argument values.

Reply:
Done.